core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -402,56 +403,37 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
         }
     }
 
-    /**
-     * Checks for version conflicts. If a non-critical version conflict exists <code>true</code> is returned. In the case of a critical
-     * version conflict (if operation origin is primary) a {@link VersionConflictEngineException} is thrown.
-     *
-     * @param op              the operation
-     * @param currentVersion  the current version
-     * @param expectedVersion the expected version
-     * @param deleted         {@code true} if the current version is not found or represents a delete
-     * @return <code>true</code> iff a non-critical version conflict (origin recovery or replica) is found otherwise <code>false</code>
-     * @throws VersionConflictEngineException if a critical version conflict was found where the operation origin is primary
-     * @throws IllegalArgumentException if an unsupported version type is used.
-     */
-    private boolean checkVersionConflict(final Operation op, final long currentVersion, final long expectedVersion, final boolean deleted) {
-        if (op.versionType() == VersionType.FORCE) {
-            if (engineConfig.getIndexSettings().getIndexVersionCreated().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
-                // If index was created in 5.0 or later, 'force' is not allowed at all
-                throw new IllegalArgumentException("version type [FORCE] may not be used for indices created after 6.0");
-            } else if (op.origin() != Operation.Origin.LOCAL_TRANSLOG_RECOVERY) {
-                // For earlier indices, 'force' is only allowed for translog recovery
-                throw new IllegalArgumentException("version type [FORCE] may not be used for non-translog operations");
-            }
-        }
-
-        if (op.versionType().isVersionConflictForWrites(currentVersion, expectedVersion, deleted)) {
-            if (op.origin() == Operation.Origin.PRIMARY) {
-                // fatal version conflict
-                throw new VersionConflictEngineException(
-                        shardId,
-                        op.type(),
-                        op.id(),
-                        op.versionType().explainConflictForWrites(currentVersion, expectedVersion, deleted));
+    enum LuceneOpStatus {
```

Comment:
The name comes from the following line of thought - maybe it helps explain it (I'm fine with any other names) - operations on replicas (where this is relevant) are always stored in the translog. They are added to lucene only if the document version *in lucene* is older than the incoming one. The enum is supposed to reflect the status of the document in lucene. Does this help? any suggestion as to how to name it to reflect it only relates the lucene index?